python/src/aiconfig/eval/README.md

```diff
@@ -0,0 +1,112 @@
+
+- You specify an AIConfig, test data, and evaluation metrics / assertions.
+- Each test case is a pair of (data, evaluation_metric).
+- The test data can either be AIConfig _inputs_ or precomputed _outputs_.
+- The evaluation metrics can be general, like string length, or have values that depend on the specific test case (like substring match, see example below).
+- Metrics are represented as floats, but can represent anything numerical: float/int, or any boolean condition (pass/fail).
+- AIConfig eval library runs them through your AIConfig (if necessary) and presents you a table to compare the results. Example:
+
+Inputs:
+
+```
+test_inputs_with_substrings = [
+    ("Food", "Magnolia Bakery"),
+    ("Iconic midtown skyscrapers", "Empire State Building"),
+]
+# For each input, ask the library to check two things:
+- substring match (given the specific paired substring)
+- brevity (string length in chars).
+...
+```
+
+In this case, we have 2 test inputs: "Food" and "Iconic midtown skyscrapers". For each input, we are checking 2 things:
+1. Does the output contain a particular substring that we expect it to? This is a boolean metric, very similar to a traditional unit test. 1.0 for pass, 0.0 for fail.
+2. Is the output brief enough. The metric is `brevity` which is just string length. Specifically, it's a string -> float function that returns the text's length in characters. This is a numerical metric, not a pass/fail test case.
+
+Output:
+
+![Example AIConfig Eval output table](image-4.png)
+
+# Usage Guide - Custom Eval
+
+## Quick start:
+
+Make sure AIConfig library is installed (pip3 install python-aiconfig).
+
+Usage: Please see [this notebook](https://github.com/lastmile-ai/aiconfig/blob/d79cf7dadf934b0ce09bc671cfde37aaecf05c1e/python/src/aiconfig/eval/custom_eval/examples/travel/travel_eval.ipynb#L4) for an example of running evaluation on an existing AIConfig. Run the notebook cells in order. This should be sufficient to get started, but the sections below provide more detail.
+
+## Custom Eval Usage Steps
+
+Let's take a closer look at what we're testing and how. The travel example linked above is a parametrized AIConfig with the following DAG topology (it is linear in this example):
+
+`the_query -> get_activities.input -> get_activities.output -> gen_itinerary.input -> gen_itinerary.output`.
+
+The leaf "the_query" is the placeholder for test case inputs, like a function argument. The root "gen_itinerary.output" is the value under test. You must specify exactly one leaf input and it must be named "the_query". You must also specify the name of the prompt you want to test (see test suite settings in the example notebook).
```

Comment:
What does the wording "is the value under test." means?

Reply:
Also for `see test suite settings in the example notebook`, we should be more specific, specifically "see `tsconfig` in the example notebook) and also link to the notebook again (just link the notebook in all places it's referenced to make it easier for the reader)